wgpu-core/src/instance.rs

```diff
@@ -59,61 +59,45 @@ pub struct Instance {
 
 impl Instance {
     pub fn new(name: &str, version: u32, backends: BackendBit) -> Self {
-        Instance {
-            #[cfg(any(
-                not(any(target_os = "ios", target_os = "macos")),
-                feature = "gfx-backend-vulkan"
-            ))]
-            vulkan: if backends.contains(Backend::Vulkan.into()) {
-                gfx_backend_vulkan::Instance::create(name, version).ok()
-            } else {
-                None
-            },
-            #[cfg(any(target_os = "ios", target_os = "macos"))]
-            metal: if backends.contains(Backend::Metal.into()) {
-                Some(gfx_backend_metal::Instance::create(name, version).unwrap())
-            } else {
-                None
-            },
-            #[cfg(windows)]
-            dx12: if backends.contains(Backend::Dx12.into()) {
-                gfx_backend_dx12::Instance::create(name, version).ok()
-            } else {
-                None
-            },
-            #[cfg(windows)]
-            dx11: if backends.contains(Backend::Dx11.into()) {
-                Some(gfx_backend_dx11::Instance::create(name, version).unwrap())
-            } else {
-                None
-            },
+        backends_map! {
+            let map = |(backend, backend_create)| {
+                if backends.contains(backend.into()) {
+                    backend_create()
+                } else {
+                    None
+                }
+            };
+            Instance {
+                #[vulkan]
+                vulkan: map((Backend::Vulkan, || gfx_backend_vulkan::Instance::create(name, version).ok())),
```

Comment:
can we just pass `gfx_backend_vulkan::Instance::create` here? It's a static function with known parameters, and map() implementation can always just call it as `backend_crate(name, version).ok()`

Reply:
So far I've preserved the semantic of this function where some of the backends panic with unwrap(), and some just silently map to None. By passing `create`, we would have to uniformly do one or the other here. If that is fine I can change it. 😄

Reply:
yeah, that's fine, let's have all of them this way